driver/main.c

```diff
@@ -1357,94 +1357,166 @@ static const unsigned char compat_nas[21] = {
 
 
 #ifdef _HAS_SOCKETCALL
-static ppm_event_code parse_socketcall(struct event_filler_arguments *filler_args, struct pt_regs *regs)
+static long convert_network_syscalls(struct pt_regs *regs)
 {
 	unsigned long __user args[6] = {};
-	unsigned long __user *scargs;
 	int socketcall_id;
 	ppm_syscall_get_arguments(current, regs, args);
 	socketcall_id = args[0];
-	scargs = (unsigned long __user *)args[1];
 
-	if (unlikely(socketcall_id < SYS_SOCKET ||
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0)
-		socketcall_id > SYS_SENDMMSG))
-#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
-		socketcall_id > SYS_RECVMMSG))
-#else
-		socketcall_id > SYS_ACCEPT4))
+	switch(socketcall_id)
+	{
+#ifdef __NR_socket
+	case SYS_SOCKET:
+		return __NR_socket;
 #endif
-		return PPME_GENERIC_E;
 
-#ifdef CONFIG_COMPAT
-	if (unlikely(filler_args->compat)) {
-		compat_ulong_t socketcall_args32[6];
-		int j;
+#ifdef __NR_socketpair
+	case SYS_SOCKETPAIR:
+		return __NR_socketpair;
+#endif
 
-		if (unlikely(ppm_copy_from_user(socketcall_args32, compat_ptr(args[1]), compat_nas[socketcall_id])))
-			return PPME_GENERIC_E;
-		for (j = 0; j < 6; ++j)
-			filler_args->socketcall_args[j] = (unsigned long)socketcall_args32[j];
-	} else {
+	case SYS_ACCEPT:
+#if defined(__TARGET_ARCH_s390) && defined(__NR_accept4)
```

Comment:
The `__TARGET_ARCH_s390` define is modern BPF specific. Should be removed from the kmod driver.

Reply:
Good point, we need to use `CONFIG_S390`

Reply:
done :)